torch/csrc/api/src/nn/init.cpp

```diff
@@ -34,8 +34,37 @@ struct Fan {
   int64_t in;
   int64_t out;
 };
+
+double calculate_kaiming_std(
+    Tensor tensor,
+    double a,
+    FanMode mode,
+    Nonlinearity nonlinearity) {
+  NoGradGuard guard;
```

Comment:
Adding `NoGradGuard` here is not mandatory because callers' guard will also apply to this function, but it's a good idea to have it here to make the intention clear.